### PR TITLE
Demo 1 refactoring

### PR DESCRIPTION
**Pull request by akurilov**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -104,11 +109,16 @@ jar {
...
     manifest {
         attributes (
-                "Implementation-Version": version,
-                "Implementation-Title": project.description,
+            "Implementation-Version": version,
+            "Implementation-Title": project.description,
         )
     }
-    
+
```
Comment (akurilov) [inline build.gradle:116]: missing statement from the HDFS storage driver, the "provided" dependencies should be packaged into the driver's jar file

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 2]
```diff
@@ -1,117 +1,147 @@
...
-		this.uriSchema = uriSchema;
-
-		final String uid = credential == null ? null : credential.getUid();
+		final String stepId, final DataInput dataInput, final Config storageConfig, final boolean verifyFlag,
+		final int batchSize
+	) throws OmgShootMyFootException {
+		super(stepId, dataInput, storageConfig, verifyFlag, batchSize);
+		this.streamConfig = StreamConfiguration
+			.builder()
+			.scalingPolicy(ScalingPolicy.fixed(concurrencyLimit))
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:195]: fixed scaling policy: count of the segments should be == concurrency

File: src/main/java/com/emc/mongoose/storage/driver/pravega/io/DataItemSerializer.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+				throw new IllegalArgumentException("Can't serialize the data item with size > 2^31 - 1");
+			}
+			if(MAX_EVENT_SIZE < dataItemSize) {
+				Loggers.ERR.warn(
+					"Event size is {}, Pravega storage doesn't support the event size more than {}",
+					SizeInBytes.formatFixedSize(dataItemSize), SizeInBytes.formatFixedSize(MAX_EVENT_SIZE)
+				);
+			}
+			if(useDirectMem) {
+				dstBuff = ByteBuffer.allocateDirect((int) dataItemSize); // will crash if not enough memory
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/io/DataItemSerializer.java:48]: actually, won't work with pravega client v0.3.2

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 4]
```diff
@@ -1,333 +1,493 @@
...
+		this.uriSchema = DEFAULT_URI_SCHEMA;
 		final Config nodeConfig = storageConfig.configVal("net-node");
 		nodePort = storageConfig.intVal("net-node-port");
 		final List<String> endpointAddrList = nodeConfig.listVal("addrs");
-		readerTimeoutMs = storageConfig.intVal("item-input-readerTimeout");
-
+		final Config driverConfig = storageConfig.configVal("driver");
+		final Config createRoutingKeysConfig = driverConfig.configVal("create-key");
+		createRoutingKeys = createRoutingKeysConfig.boolVal("enabled");
+		createRoutingKeysPeriod = createRoutingKeysConfig.longVal("count");
```
Comment (veronikaKochugova) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:204]: Why the period is called "count" ?

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 5]
```diff
@@ -1,333 +1,493 @@
...
+	}
+
+	void submitStreamOperation(final PathOperation streamOp, final OpType opType) {
+		final String nodeAddr = streamOp.nodeAddr();
+		switch(opType) {
+			case CREATE:
+				submitCreateStreamOperation(streamOp, nodeAddr);
+				break;
+			case READ:
+				throw new AssertionError("Not implemented");
```
Comment (veronikaKochugova) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java]: Why an error? I thought we could submit his operation

**Review by veronikaKochugova — APPROVED** [review]

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java (hunk 2 above)
Comment (veronikaKochugova) [reply src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:195]: Does pravega guarantee that there will be as many segments as we set? Or they may be less?
Just a limit usually means conc <= concLimit but not conc = concLimit

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java (hunk 4 above)
Comment (akurilov) [reply src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:204]: an user may incorrectly consider a period as something measured in time units

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java (hunk 5 above)
Comment (akurilov) [reply src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java]: yep, fixed

**Review by dadlex — COMMENTED** [review]
All the refactoring seems appropriate

**Review by igkon — COMMENTED** [review]
everything looks fine

**Pull request merged**